Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts

```diff
@@ -1,65 +1,120 @@
         this.getTestSettings(this.testId);
     }
 
     //Gets the Settings saved for a particular Test
-    getTestSettings(id:number) {
+    /**
+     * Gets the Settings saved for a particular Test
+     * @param id contains the value of the Id from the route
+     */
+    getTestSettings(id: number) {
         this.testSettingService.getSettings(id).subscribe((response) => {
             this.testsettings = (response);
         });
     }
 
-    //Selects the Test Name from the text box containing it on focus 
+    /**
+     * Selects the Test Name from the text box containing it on focus
+     * @param $event is used to select the contents of the target text box
+     */
     selectAllContent($event: any) {
         $event.target.select();
-    }   
+    }
 
-    //Updates the Edited Test Name in the Database on removing focus
-    onBlurMethod(id: number, testObject: Test) {
-        this.testSettingService.updateSettings(id, testObject).subscribe((response) => {       
+    /**
+     * Updates the Edited Test Name in the Database om removing focus
+     * @param id contains the value of the Id from the route
+     * @param testObject is an object of class Test
+     * @param value contains the Test Name of the Test which is to be edited 
+     */
+    testNameUpdation(id: number, testObject: Test, value: string) {
+        this.testSettingService.updateSettings(id, testObject).subscribe((response) => {
         });
     }
 
-    //Launches the Test Dialog Box and also updates the Settings edited for the selected Test
+    /**
+     * Checks the End Date and Time is valid or not
+     * @param endDate contains ths the value of the field End Date and Time
+     */
+    validEndDateChecking(endDate: Date) {
+        if (this.testsettings.startDate > endDate)
+            this.validEndDate = true;
+        else
+            this.validEndDate = false;
+    }
+
+    /**
+     * Checks whether the Start Date selected is valid or not
+     */
+    validStartDateChecking() {
+        if (new Date(this.testsettings.startDate) < this.currentDate || this.testsettings.startDate > this.testsettings.endDate)
+            this.validStartDate = true;
+        else
+            this.validStartDate = false;
+    }
+
+    /**
+     * Checks whether the Warning Time set is valid
+     */
+    validWarningTimeChecking() {
+        if (this.testsettings.warningTime >= this.testsettings.duration)
+            this.validTime = true;
+        else
+            this.validTime = false;
+    }
+
+    /**
+     * Launches the Test Dialog Box and also updates the Settings edited for the selected Test
+     * @param id contains the value of the Id from the route
+     * @param testObject is an object of class Test
+     */
     launchTestDialog(id: number, testObject: Test) {
-        this.testSettingService.updateSettings(id, testObject).subscribe((response) => {         
+        this.testSettingService.updateSettings(id, testObject).subscribe((response) => {
```

Comment:
I know what is happening on server side. My question is after update what are you doing to let user know that data is updated?